src/Items/ItemFishingRod.h

```diff
@@ -244,9 +244,9 @@ class cItemFishingRodHandler :
 		}
 		else
 		{
-			cFloater * Floater = new cFloater(a_Player->GetPosX(), a_Player->GetStance(), a_Player->GetPosZ(), a_Player->GetLookVector() * 15, a_Player->GetUniqueID(), static_cast<int>(100 + static_cast<unsigned int>(a_World->GetTickRandomNumber(800)) - (a_Player->GetEquippedItem().m_Enchantments.GetLevel(cEnchantments::enchLure) * 100)));
-			Floater->Initialize(*a_World);
+			auto Floater = cpp14::make_unique<cFloater>(a_Player->GetPosX(), a_Player->GetStance(), a_Player->GetPosZ(), a_Player->GetLookVector() * 15, a_Player->GetUniqueID(), static_cast<int>(100 + static_cast<unsigned int>(a_World->GetTickRandomNumber(800)) - (a_Player->GetEquippedItem().m_Enchantments.GetLevel(cEnchantments::enchLure) * 100)));
 			a_Player->SetIsFishing(true, Floater->GetUniqueID());
+			Floater->Initialize(std::move(Floater), *a_World);
```

Comment:
This order reversal is dangerous, what if `SetIsFishing` was rewritten to actually do something with the floater's ID, searching the world for it for example? I think the initialization should come first.